### DISCUSSION TITLE
Update deprecated gradle methods

### DISCUSSION THREAD

**Discussion by ryan-conway** [nimblehq/android-templates#439] [RFCs]
## Why

`lintOptions` has been deprecated since Gradle 7.1 and `flavorDimensions` has been deprecated since Gradle 7.2. Since we are currently using Gradle 7.3, we should update these with the appropriate replacements.

## Solution

- Replace `lintOptions` block with `lint`
- Replace `flavorDimensions` method with `flavorDimensions` property

## Who Benefits?

Android developers

**Comment by luongvo** [comment]
@ryan-conway These updates are so clear and straight. I don't think we really need this RFC.

**Reply by toby-thanathip** [reply]
@luongvo I agree, but this is what we voted for in #417.

I guess we could make an exception for issues that are "simple" and "straightforward" 💭 

I'm just worried that what is "straightforward" and what is not, can be subjective depending on the experience of the developer.

So IMO, it might be safer and easier to create an RFC for everything.

@ryan-conway What do you think?

**Reply by ryan-conway** [reply]
Agreed that "straightforward" doesn't mean the same thing to everyone. If we make exceptions for some issues, where is the line between those issue and ones that need votes? For example, this issue may seem fairly simple to handle but the team may feel that it isn't worth the effort (however small) to implement it, therefore we may reject it for now.

I think since we voted in #417, that is how we should handle things. Otherwise the decision there is meaningless IMO 🤔
